merlin-server/sql/merlin/functions/public/restore_from_snapshot.sql

```diff
@@ -0,0 +1,102 @@
+create procedure restore_from_snapshot(_plan_id integer, _snapshot_id integer)
+  language plpgsql as $$
+  declare
+    _snapshot_name text;
+    _plan_name text;
+  begin
+    -- Input Validation
+    select name from plan where id = _plan_id into _plan_name;
+    if _plan_name is null then
+      raise exception 'Cannot Restore: Plan with ID % does not exist.', _plan_id;
+    end if;
+    if not exists(select snapshot_id from plan_snapshot where snapshot_id = _snapshot_id) then
+      raise exception 'Cannot Restore: Snapshot with ID % does not exist.', _snapshot_id;
+    end if;
+    if not exists(select snapshot_id from plan_snapshot where _snapshot_id = snapshot_id and _plan_id = plan_id ) then
+      select snapshot_name from plan_snapshot where snapshot_id = _snapshot_id into _snapshot_name;
+      if _snapshot_name is not null then
+				raise exception 'Cannot Restore: Snapshot ''%'' (ID %) is not a snapshot of Plan ''%'' (ID %)',
+				  _snapshot_name, _snapshot_id, _plan_name, _plan_id;
+			else
+        raise exception 'Cannot Restore: Snapshot % is not a snapshot of Plan ''%'' (ID %)',
+				  _snapshot_id, _plan_name, _plan_id;
+			end if;
+		end if;
```

Comment:
Weird formatting on this `end if` makes it seem like this block is scoped differently. Alternatively, we could maybe unnest these `if`s and just have a flat list of checks / exception raises.

Reply:
Oh, yes, there's something weird where GitHub is drawing the formatting wrong. If you open the file locally, you shouldn't see this. I will update the formatting here and bring it all onto space formatting 👍

Reply:
classic tabs vs spaces

Reply:
Formatting has been fixed